basecli/k8s.go

```diff
@@ -128,12 +135,33 @@ iter8 gen k8s`,
 	},
 }
 
+// GetIdFlag returns the id flag.
+// This function enables reuse of this flag across subcommands.
+func GetIdFlag() *pflag.Flag {
+	name := "id"
+	f := pflag.Lookup(name)
+	if f == nil {
+		pflag.StringVarP(&Id, name, "i", "", "if not specified, a randomly generated identifier will be used")
+	}
+	return pflag.Lookup(name)
```

Comment:
Why this pattern? It seems more complex and I am inclined to think we want to observe and error if the parameter already exists. Otherwise, we may miss a conflict.

Reply:
Order in which init() functions are called in go is non-deterministic. 

So,  there is no real way  to  make sure the flag is added in `k8s.go` before it can be retrieved in `iter8 k assert.go` or `delete.go` for instance.

This pattern makes the order of init invocations irrelevant. If the flag is not already part of the default `pflag` flagset, it will simply be added and then returned, which is the desired behavior.

Reply:
I do not understand how the flag being added in `k8s.go` and in `delete.go` are related to each other. These are different cobra `Command`s. Qhy is the order of `init()` between them relevant?

Reply:
Both `iter8 gen k8s` and `iter8 k delete/assert/...` make use of `id` and `app` variables and flags.

The pattern above ensures that:
1. App and Id variables are defined in place and reused across these subcommands.
2. Flags are available for these subcommands to set these variables through command line.
3. The flags themselves are reused. In other words, there is a single pflag.Flag object, included as part of the pflag.Flagset, and looked up by whoever needs it.
4. Both the initial addition to the flagset and the lookups happen as part of the init functions. But there is no way to control the order of init function executions. This means, you cannot simply place the addition of the flag to the flagset in one init func, and expect to see it present in another init func. Thee pattern used above ensures that no matter which init func is called first, the flag will be added to the flagset, once and exactly once,  and will be reused.

Reply:
Similar to the idea of using NewCmd() patterns for reusing command blocks, this also reuses the flags passed to those command blocks.

Reply:
If you do have a simple pattern to achieve 1, 2, 3 and 4 above -- we can certainly consider it!

Reply:
The following alternative should work as well... but its a bit of careless coding, though simpler! This means, the app flag is recreated in the pflag flagset, every time any command "gets" it, which seems... wasteful. I don't really think there is a big diff either way.

```go
 // GetAppFlag returns the app flag.
 // This function enables reuse of this flag across subcommands.
 func GetAppFlag() *pflag.Flag {
 	pflag.StringVarP(&App, name, "a", "", "label to be associated with an experiment, default is 'default'")
 	return pflag.Lookup(name)
 }

Reply:
@kalantar I now understand that the issue is a bit more deeper than docs and code structure ... 

1. `iter8 gen k8s` current does not (rightly so) ignore either id or app when they are specified. However `iter8 k *` commands current do ignore app when id is specified. So far so good.
2. This means the app flags are actually different across these two command groups (because the behavior of the app flag is different).
3. This means, we don't need `GetAppFlag`. `GetIdFlag` still makes sense, and the way App and Id variables are reused still makes sense.

I will push these changes shortly.

Reply:
Fixed now.